server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesUpdateProcessor.java

```diff
@@ -121,15 +153,29 @@ private void update(MessageId messageId, UpdateMessagePatch updateMessagePatch,
 
             handleInvalidRequest(builder, messageId, ImmutableList.of(invalidPropertyKeywords));
         }
-
     }
 
+
     private Stream<MailboxId> listTargetMailboxIds(UpdateMessagePatch updateMessagePatch) {
         return OptionalUtils.toStream(updateMessagePatch.getMailboxIds())
             .flatMap(Collection::stream)
             .map(mailboxIdFactory::fromString);
     }
 
+    private boolean isTargetingOutbox(MailboxSession mailboxSession, Stream<MailboxId> targetMailboxIds) throws MailboxException {
+        Set<MailboxId> outboxes = listOutboxMailboxes(mailboxSession);
+        if (outboxes.isEmpty()) {
+            throw new MailboxNotFoundException("At least one outbox should be accessible");
```

Comment:
From the method naming you are not targetting outbox (because none exists), you should be returning false.

I find that exception misleading.

Reply:
I think that's an error case : your account doesn't have any outbox. It's what exception are for, exceptional situation, right ?

Reply:
interesting question. I think it depends. For now every account should have an outbox. But what if we introduce functional mailboxes as accounts without mailbox?
When it's not harmful, I have a slight preference in ignoring such errors (just log them).
Of course in other cases where we want to ensure some business rules, we should throw at the first encountered error.

Reply:
In this case, I agree with benoit that the name implies that if you don't have any outbox, it can't target outbox and thus return false.